app/code/Magento/Inventory/Model/SourceItemRepository.php

```diff
@@ -0,0 +1,163 @@
+{
+    /**
+     * @var ResourceSource
+     */
+    private $resourceSource;
+
+    /**
+     * @var SourceItemInterfaceFactory
+     */
+    private $sourceItemFactory;
+
+    /**
+     * @var CollectionProcessorInterface
+     */
+    private $collectionProcessor;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $sourceItemCollectionFactory;
+
+    /**
+     * @var SourceItemSearchResultsInterfaceFactory
+     */
+    private $sourceItemSearchResultsFactory;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * SourceRepository constructor
+     *
+     * @param ResourceSource $resourceSource
+     * @param SourceItemInterfaceFactory $sourceItemFactory
+     * @param CollectionProcessorInterface $collectionProcessor
+     * @param CollectionFactory $sourceItemCollectionFactory
+     * @param SourceItemSearchResultsInterfaceFactory $sourceItemSearchResultsFactory
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ResourceSource $resourceSource,
+        SourceItemInterfaceFactory $sourceItemFactory,
+        CollectionProcessorInterface $collectionProcessor,
+        CollectionFactory $sourceItemCollectionFactory,
+        SourceItemSearchResultsInterfaceFactory $sourceItemSearchResultsFactory,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        LoggerInterface $logger
+    ) {
+        $this->resourceSource = $resourceSource;
+        $this->sourceItemFactory = $sourceItemFactory;
+        $this->collectionProcessor = $collectionProcessor;
+        $this->sourceItemCollectionFactory = $sourceItemCollectionFactory;
+        $this->sourceItemSearchResultsFactory = $sourceItemSearchResultsFactory;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function save(array $sourceItemList)
+    {
+        try {
+            $sourceItemData = [];
+
+            /** @var SourceItemInterface $sourceItem */
+            foreach ($sourceItemList as $sourceItem) {
+                $sourceItemData[] = [
+                    SourceItemInterface::SOURCE_ITEM_ID => $sourceItem->getSourceItemId(),
+                    SourceItemInterface::SOURCE_ID => $sourceItem->getSourceId(),
+                    SourceItemInterface::SKU => $sourceItem->getSku(),
+                    SourceItemInterface::QUANTITY => $sourceItem->getQuantity(),
+                    SourceItemInterface::STATUS => $sourceItem->getStatus()
+                ];
+            }
+
+            $connection = $this->resourceSource->getConnection();
+
+            $connection->insertMultiple(
+                $connection->getTableName(InstallSchema::TABLE_NAME_SOURCE_ITEM),
+                $sourceItemData
```

Comment:
insertMultiple - should be done by Resource model, not Repository.

The repository should be agnostic to the precise adapter which is used under the hood.
So, switching MySQL -> Postgres SQL -> Mongo DB should not affect Repository code. Just persistent layer should be affected (in our current conditions it's Resource model).

In your implementation Repository works directly with MySQL connection